transport/src/main/java/io/netty/channel/socket/nio/NioSocketChannel.java

```diff
@@ -380,8 +381,15 @@ protected int doWriteBytes(ByteBuf buf) throws Exception {
 
     @Override
     protected long doWriteFileRegion(FileRegion region) throws Exception {
-        final long position = region.transferred();
-        return region.transferTo(javaChannel(), position);
+        if (region instanceof SliceableFileRegion) {
+            SliceableFileRegion r = (SliceableFileRegion) region;
+            return r.transferBytesTo(javaChannel(), r.transferableBytes());
+        } else {
+            // TODO: consider remove this branch once we move the methods in SliceableFileRegion
```

Comment:
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png) Complete the task associated to this TODO comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3AS1135)